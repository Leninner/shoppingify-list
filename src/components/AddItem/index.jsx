import { StyledAddItemContainer, BoxInput, Form, ButtonAddItem } from './styles';

export const AddItem = () => {
  return (
    <StyledAddItemContainer>
      <Form>
        <BoxInput type='text' name='item' placeholder='Enter a name' disabled />
        <ButtonAddItem disabled>Save</ButtonAddItem>
      </Form>
    </StyledAddItemContainer>
  );
};
