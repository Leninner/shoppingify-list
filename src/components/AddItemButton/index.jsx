import {
  StyledAddItemContainer,
  BoxInput,
  Form,
  ButtonAddItem,
  ButtonConfirm,
  Div,
  ButtonCompletingList,
} from './styles';
import { useSelector, useDispatch } from 'react-redux';

export const AddItemButton = ({
  isAdding,
  handleAddItem,
  isItemInfo,
  idItem,
  DeleteItemFromCategories,
  categoryName,
  idCategory,
  AddToCurrentList,
  itemName,
  AddToHistory,
}) => {
  const { shoppingCartReducer } = useSelector((state) => state);

  // console.log({
  //   shoppingCartReducer,
  //   historialReducer,
  //   categoriesReducer,
  // });

  const { shoppingCart, isCompleting } = shoppingCartReducer;

  const dispatch = useDispatch();

  const handleClick = (e) => {
    e.preventDefault();
    handleAddItem();
  };

  const handleDeleteItemFromCategories = () => {
    DeleteItemFromCategories({ idItem, categoryName, idCategory, itemName });
  };

  const handleAddToCurrentList = () => {
    AddToCurrentList({ idItem, categoryName, idCategory });
  };

  const isCurrentListEmpty = shoppingCart.every((category) => category.items.length === 0);

  const isAddedToCurrentList = shoppingCart.some((category) =>
    category.items.some((item) => {
      return item.id === idItem && item.itemName === itemName ? true : false;
    })
  );

  const handleCancelDeleteList = () => {
    dispatch({
      type: 'CANCEL_DELETE_LIST',
    });
  };

  const handleCompleteList = () => {
    console.log(AddToHistory);
    AddToHistory();
  };

  return (
    <StyledAddItemContainer>
      {!isCompleting ? (
        <>
          {!isItemInfo ? (
            <>
              {isAdding ? (
                <Div>
                  <ButtonConfirm isCancel onClick={handleClick} type='button'>
                    Cancel
                  </ButtonConfirm>
                  <ButtonConfirm type='submit'>Save</ButtonConfirm>
                </Div>
              ) : (
                <Form>
                  <BoxInput type='text' name='item' placeholder='Enter a name' disabled={isCurrentListEmpty} />
                  <ButtonAddItem disabled={isCurrentListEmpty}>Save</ButtonAddItem>
                </Form>
              )}
            </>
          ) : (
            <Div>
              <ButtonConfirm isCancel onClick={handleDeleteItemFromCategories}>
                Delete
              </ButtonConfirm>
              {isAddedToCurrentList ? null : (
                <ButtonConfirm onClick={handleAddToCurrentList}>Add to list</ButtonConfirm>
              )}
            </Div>
          )}
        </>
      ) : (
        <Div>
          <ButtonCompletingList isCancelList onClick={handleCancelDeleteList}>
            Cancel
          </ButtonCompletingList>
          <ButtonCompletingList onClick={handleCompleteList}>Complete</ButtonCompletingList>
        </Div>
      )}
    </StyledAddItemContainer>
  );
};
