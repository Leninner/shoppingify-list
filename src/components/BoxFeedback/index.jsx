import { StyledBoxFeedback, Botella, Info, Button, Description } from './styles';
import botella from '../../assets/source.svg';

export const BoxFeedback = ({ handleAddItem }) => {
  const handleClick = () => {
    handleAddItem(true);
  };

  return (
    <StyledBoxFeedback>
      <Botella src={botella} alt='botella' />
      <Info>
        <Description>Didnâ€™t find what you need?</Description>
        <Button onClick={handleClick}>Add Item</Button>
      </Info>
    </StyledBoxFeedback>
  );
};
