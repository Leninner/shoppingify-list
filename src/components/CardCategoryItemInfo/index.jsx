import { StyledCardCategoryItemInfo, Return, ImgItem, ContainerInfo } from './styles';

const imageDefault =
  'https://images.unsplash.com/photo-1523049673857-eb18f1d7b578?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=775&q=80';

export const CardCategoryItemInfo = ({ children, itemName, image }) => {
  return (
    <StyledCardCategoryItemInfo>
      <Return onClick={() => console.log('Back')}>
        <span className='material-icons'>arrow_back</span> Back
      </Return>

      <ImgItem src={image || imageDefault} alt='' />

      <ContainerInfo>
        <span>name</span>
        <h3>{itemName}</h3>
      </ContainerInfo>

      <ContainerInfo>
        <span>category</span>
        <p>Fruit And Vegetables</p>
      </ContainerInfo>

      <ContainerInfo>
        <span>note</span>
        <p>
          Nutrient-dense foods are those that provide substantial amounts of vitamins, minerals and other nutrients with
          relatively few calories. One-third of a medium avocado (50 g) has 80 calories and contributes nearly 20
        </p>
      </ContainerInfo>
      {children}
    </StyledCardCategoryItemInfo>
  );
};
