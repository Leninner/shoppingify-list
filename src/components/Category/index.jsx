import { ItemCategory } from '../ItemCategory';
import { TitleCategory, StyledCategoryContainer } from './styles';

export const Category = (category) => {
  const { name, items } = category;

  return (
    <StyledCategoryContainer>
      <TitleCategory>{name}</TitleCategory>
      {items.map((item) => (
        <ItemCategory key={item.id} {...item} />
      ))}
    </StyledCategoryContainer>
  );
};
