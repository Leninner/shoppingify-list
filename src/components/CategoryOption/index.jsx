import { Option } from './styles';

export const CategoryOptions = ({ category, onClick }) => {
  return (
    <Option onClick={onClick}>
      <span>{category}</span>
    </Option>
  );
};
