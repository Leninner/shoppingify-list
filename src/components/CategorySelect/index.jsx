import { CategoryOptions } from '../CategoryOption';
import { ContainerOptions, Container, Close } from './styles';
import PropTypes from 'prop-types';
import { useNearScreen } from '../../hooks/useNearScreen';

export const CategorySelect = ({
  WrapperInput,
  categories,
  isOpen,
  setIsOpen,
  selected,
  setSelected,
  disabled,
  setDisabled,
}) => {
  const elementRef = useNearScreen(isOpen);

  const categoriesFiltered = categories.filter((category) =>
    category.name.toLowerCase().includes(selected.toLowerCase().trim())
  );

  const handleChange = (e) => {
    setSelected(e.target.value);
  };

  const handleClick = (category) => {
    setSelected(category);
    setDisabled(true);
    setIsOpen(false);
  };

  const handleCancel = () => {
    setIsOpen(false);
    setDisabled(false);
    setSelected('');
  };

  return (
    <Container>
      <WrapperInput
        type='text'
        placeholder='Select a category or create a new one'
        name='category'
        autoComplete='off'
        value={selected}
        onChange={handleChange}
        onClick={() => setIsOpen(true)}
        disabled={disabled}
      />

      {selected && (
        <Close className='material-icons' onClick={handleCancel}>
          close
        </Close>
      )}

      {isOpen && (
        <ContainerOptions ref={elementRef}>
          {categoriesFiltered.map((elemento) => {
            const { id, name } = elemento;
            return <CategoryOptions key={id} category={name} onClick={() => handleClick(name)} />;
          })}

          {!categoriesFiltered.length && (
            <CategoryOptions category={`+ AÃ±adir ${selected}`} onClick={() => handleClick(selected)} />
          )}
        </ContainerOptions>
      )}
    </Container>
  );
};

CategorySelect.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.object).isRequired,
};
