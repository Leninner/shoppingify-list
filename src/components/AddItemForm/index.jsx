import { StyledAddItemForm, Campo, Entrada, EntradaNote, Section, Error } from './styles';
import { CategorySelect } from '../CategorySelect';
import { useFormik } from 'formik';

export const AddItemForm = ({ categories, children }) => {
  const validate = (values) => {
    const errors = {};

    if (!values.itemName || values.itemName.trim() === '') {
      errors.itemName = 'Reguired';
    }

    return errors;
  };

  const formik = useFormik({
    initialValues: {
      itemName: '',
      note: '',
      image: '',
      category: '',
    },
    onSubmit: (values) => {
      console.log(values);
    },
    validate,
  });

  return (
    <StyledAddItemForm onSubmit={formik.handleSubmit}>
      <h3>Add a new item</h3>
      <Section>
        <Campo htmlFor='itemName'>
          <span>Name *</span>
          <Entrada
            type='text'
            name='itemName'
            placeholder='Enter a name'
            onChange={formik.handleChange}
            value={formik.values.itemName}
            onBlur={formik.handleBlur}
          />
          {formik.errors.itemName && <Error>{formik.errors.itemName}</Error>}
        </Campo>
        <Campo>
          <span>Note (optional)</span>
          <EntradaNote
            name='note'
            placeholder='Enter a note'
            onChange={formik.handleChange}
            value={formik.values.note}
            onBlur={formik.handleBlur}
          />
        </Campo>
        <Campo>
          <span>Image (optional)</span>
          <Entrada
            type='url'
            name='url'
            placeholder='Enter a url'
            onChange={formik.handleChange}
            value={formik.values.image}
            onBlur={formik.handleBlur}
          />
        </Campo>
        <Campo>
          <span>Category *</span>
          <CategorySelect WrapperInput={Entrada} categories={categories} />
        </Campo>

        {children}
      </Section>
    </StyledAddItemForm>
  );
};
