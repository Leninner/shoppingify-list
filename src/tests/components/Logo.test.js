import { LogoComponent } from '../../components/Logo';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

describe('Debe estar presente en el DOM', () => {
  const logo = shallow(<LogoComponent />);

  test('Debe estar presente un SVG', () => {
    expect(logo).toMatchSnapshot();
  });
});
