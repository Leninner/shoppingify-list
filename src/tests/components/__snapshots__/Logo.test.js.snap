// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe estar presente en el DOM Debe estar presente un SVG 1`] = `
<styled.svg
  fill="none"
  height={43}
  width={42}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M31.57 2.365c.514 0 .93-.418.93-.933a.932.932 0 0 0-.93-.933.932.932 0 0 0-.93.933c0 .515.416.933.93.933ZM31.57 9.746c.514 0 .93-.417.93-.933a.932.932 0 0 0-.93-.933.932.932 0 0 0-.93.933c0 .516.416.933.93.933ZM40.871 11.613c.514 0 .93-.418.93-.933a.932.932 0 0 0-.93-.934.932.932 0 0 0-.93.934c0 .515.417.933.93.933Z"
    fill="#3F3D56"
  />
  <path
    d="M39.342 14.765c-.388-.643-1.745-.626-1.745-.626-1.236 1.842-4.205-.114-4.205-.114-2.205.99-3.551-2.165-3.551-2.165-2.968-.446-2.51-3.573-2.51-3.573-1.476-.297-1.061-2.976-1.061-2.976s.819-.674.416-1.167c-.257-.314-.273-1.214-.25-1.828a20.311 20.311 0 0 0-5.944-.884C9.239 1.432.114 10.586.114 21.879s9.124 20.447 20.378 20.447c11.255 0 20.38-9.154 20.38-20.447a20.462 20.462 0 0 0-1.295-7.184c-.105.118-.186.15-.235.07Z"
    fill="#3F3D56"
  />
  <path
    d="M38.134 14.685c-.37-.612-1.66-.596-1.66-.596-1.174 1.752-3.998-.109-3.998-.109-2.098.942-3.378-2.059-3.378-2.059-2.822-.423-2.386-3.398-2.386-3.398-1.404-.282-1.01-2.83-1.01-2.83s.779-.64.395-1.11c-.243-.298-.258-1.154-.237-1.738a19.317 19.317 0 0 0-5.652-.841C9.503 2.004.825 10.71.825 21.45c0 10.74 8.678 19.446 19.381 19.446 10.704 0 19.382-8.706 19.382-19.446a19.458 19.458 0 0 0-1.232-6.832c-.1.112-.176.143-.223.067Z"
    fill="#3F3D56"
  />
  <path
    d="M17.893 13.5s-3.342-2.134-6.532 0c-1.117.747-1.675 1.81-1.929 2.879-.452 1.899.044 3.901 1.251 5.432l7.218 9.147a2.146 2.146 0 0 0 2.186.771 2.148 2.148 0 0 0 1.36-1.036l5.02-8.844a8.59 8.59 0 0 0 1.084-5.328c-.235-1.819-1.106-3.736-3.581-4.393 0 0-2.886-1.22-6.077 1.372Z"
    fill="#F9A109"
  />
  <path
    d="M17.893 13.5s-3.342-2.134-6.532 0c-1.117.747-1.675 1.81-1.929 2.879-.452 1.899.044 3.901 1.251 5.432l7.218 9.147a2.146 2.146 0 0 0 2.186.771 2.148 2.148 0 0 0 1.36-1.036l5.02-8.844a8.59 8.59 0 0 0 1.084-5.328c-.235-1.819-1.106-3.736-3.581-4.393 0 0-2.886-1.22-6.077 1.372Z"
    fill="#000"
    opacity={0.1}
  />
  <path
    d="M17.927 14.015s-3.135-2.002-6.127 0c-1.048.7-1.572 1.699-1.81 2.7-.424 1.782.041 3.66 1.174 5.096l6.77 8.58a2.013 2.013 0 0 0 2.051.724 2.015 2.015 0 0 0 1.276-.972l4.709-8.296a8.058 8.058 0 0 0 1.017-4.998c-.221-1.706-1.038-3.505-3.36-4.121 0 0-2.707-1.144-5.7 1.287Z"
    fill="#F9A109"
  />
  <path
    d="M18.05 17.447s-1.526-.974-2.982 0a2.113 2.113 0 0 0-.88 1.314 2.927 2.927 0 0 0 .57 2.48s3.906 4.613 4.271 4.531l.022-.005c.263-.063 2.911-4.51 2.911-4.51a3.92 3.92 0 0 0 .495-2.43c-.107-.83-.505-1.706-1.634-2.006 0 0-1.318-.556-2.774.626ZM8.522 24.739l.427.715.855 1.43.285 1-.57.858.713 1.716.285 1.001-.57.572 1.14.858M29.47 18.447v1.144l-.712 1.859-.428 1 .713 1.288.285.858 1.14.429.428.715v1.716M36.739 19.591l-1.14.715s-.57.286-.57.43v1.143M31.608 15.588l.285 1.43-.285 1M32.748 28.456l-.855 1.287-.997.858v.572l.57.715v.858l-.428.858M24.625 4.149H23.2l-1.14 1-1.852 1.001-.856.715v.286L17.5 8.152l-.57 1.001M9.234 6.58 7.952 9.152l-.428 1-.997 1.288s-.713 1.287-.855 1.43c-.143.143-1.14.429-1.14.429l1.425 1.43M3.534 16.445l-.285 1.86.285 1.143v2.002l-.855 3.718.855 1 .998 1.288M5.672 17.303l.855.43.57 1.858 1.14-.286.57 1.001M21.348 35.892l2.28-.286.997-1.001h1.425M7.524 32.889l.855 1 1.425.43.998.286.285 1.573M24.34 28.027l.998-.715.427-1.143.855-.143.428-1.43"
    fill="#000"
    opacity={0.1}
  />
</styled.svg>
`;
